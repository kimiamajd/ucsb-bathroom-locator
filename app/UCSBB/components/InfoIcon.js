import React from 'react';
import { FontAwesome } from '@expo/vector-icons';

import Colors from '../constants/Colors';

export default function InfoIcon(props) {
  return (
    <FontAwesome
      name={props.name}
      size={26}
      /*style={{ alignContent: 'left' }}*/
      color={Colors.infoIcon}
    />
  );
}
