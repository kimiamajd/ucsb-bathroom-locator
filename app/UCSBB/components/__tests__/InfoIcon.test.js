import React from 'react';
import InfoIcon from '../InfoIcon';
import TestRenderer from 'react-test-renderer';

describe ('Info Icon', () => {
    it('should render the Male icon and match to snapshot', () => {
        const component = TestRenderer.create(<InfoIcon name='male'/>).toJSON(); //shallow(<InfoIcon name='male'/>)
        expect(component).toMatchSnapshot();
    });

    it('should render the Female icon and match to snapshot', () => {
        const component = TestRenderer.create(<InfoIcon name='female'/>).toJSON(); //shallow(<InfoIcon name='female'/>)
        expect(component).toMatchSnapshot();
    });

    it('should render the Neutral icon and match to snapshot', () => {
        const component = TestRenderer.create(<InfoIcon name='user'/>).toJSON(); //shallow(<InfoIcon name='user'/>)
        expect(component).toMatchSnapshot();
    });

    it('should render the Wheelchair icon and match to snapshot', () => {
        const component = TestRenderer.create(<InfoIcon name='wheelchair'/>).toJSON(); //shallow(<InfoIcon name='wheelchair'/>)
        expect(component).toMatchSnapshot();
    });
});
