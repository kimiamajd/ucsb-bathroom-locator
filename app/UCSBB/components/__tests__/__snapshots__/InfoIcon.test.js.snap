// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info Icon should render the Female icon and match to snapshot 1`] = `<Text />`;

exports[`Info Icon should render the Male icon and match to snapshot 1`] = `<Text />`;

exports[`Info Icon should render the Neutral icon and match to snapshot 1`] = `<Text />`;

exports[`Info Icon should render the Wheelchair icon and match to snapshot 1`] = `<Text />`;
